File: 2nd-try/541/go/541.go
package main

import "fmt"

func reverse(s string, begin int, end int) string {
	r := []rune(s)
	for i, j := begin, end-1; i < j; i, j = i+1, j-1 {
		r[i], r[j] = r[j], r[i]
	}
	s = string(r)
	return s
}

func reverseStr(s string, k int) string {
	for i := 0; i < len(s); i += 2 * k {
		if len(s)-i >= k {
			s = reverse(s, i, i+k)
		} else {
			s = reverse(s, i, len(s))
		}
	}
	return s
}

func main() {
	s := "abcdefg"
	s = reverseStr(s, 2)
	fmt.Println("result:", s)
}

File: 2nd-try/541/541.cpp
#include <iostream>
#include <algorithm>

using namespace std;

class Solution
{
public:
    string reverseStr(string s, int k)
    {
        for (int i = 0; i < s.length(); i += 2 * k) {
            if (s.length() - i >= k) // 后面剩的够k个就反转k个
                reverse(s.begin() + i, s.begin() + i + k);
            else // 否则把剩的全部反转
                reverse(s.begin() + i, s.end());
        }
        return s;
    }
};

int main() {
    string s = "abcdefg";
    Solution obj;
    string res = obj.reverseStr(s, 2);
    cout << res << endl;

    return 0;
}